### ISSUE TITLE
Left alignment of status icons in Models tables

### ISSUE THREAD

**Issue by spencerbygraves** [canonical-web-and-design/jaas-dashboard#339]
@clagom commented on [Thu Jan 30 2020](https://github.com/canonical-web-and-design/juju-squad/issues/1092)

@spencerbygraves could you QA and give the specs for the correct left aligment by design?

![Screenshot 2020-01-30 at 14.06.00.png](https://images.zenhubusercontent.com/5b0d889d89d5dc41604fa14c/a01c531d-93e0-452a-ace8-3f2a386eff1a)

---

@spencerbygraves commented on [Thu Jan 30 2020](https://github.com/canonical-web-and-design/juju-squad/issues/1092#issuecomment-580362007)

[This design](https://zpl.io/beOAPkq) shows how the status icons are aligned left, and the table headings aligned with the text below.

![image](https://user-images.githubusercontent.com/9105999/73473593-eed67e80-4384-11ea-984b-f8ba1e67312c.png)

**Comment by spencerbygraves** [comment]
@hatched related to this, @clagom mentioned that the alignment of the status column on the details page should be the same. Maas also uses this alignment of icon before text.

![image](https://user-images.githubusercontent.com/9105999/73663224-0ec3b600-4695-11ea-841a-72e3b8d18c4c.png)


Can you look at that as part of this?

**Comment by barrymcgee** [comment]
@spencerbygraves These tables are generated using the MainTable component from the React Vanilla Library - has this change been considered upstream or discussed in Vanilla WG?

**Comment by spencerbygraves** [comment]
@barrymcgee it features in @lyubomir-popov's [designs for the modular table component](https://zpl.io/amvLOPK), discussed with @clagom. [It's being documented at the moment](https://discourse.ubuntu.com/c/design/modular-table-component).

I'm not sure what's in the current react table. Where does that leave us with this?

**Comment by barrymcgee** [comment]
@spencerbygraves As a general rule, we should not be speculatively building stuff based on an assumption it will eventually be part of Vanilla but rather implement the required changes as and when they reach us in the current version of Vanilla.

However, if this is a pressing priority, we can explore moving the status icon into it's own column before the status value column?

**Comment by spencerbygraves** [comment]
@barrymcgee Ok, thanks. I think we can wait on this one.

**Comment by lyubomir-popov** [comment]
@barrymcgee Just a headsup, in case you get back to this before it is in vanilla.. As part of the design stage I did some preliminary tests, with icons next to icons inside ghost buttons. I came to the conclusion that a separate column will not allow consistent spacing regardless of whether the white space is inside the element (as padding) or outside (as margin).  So a more versatile option could be a placeholder or a utility that reserves icon space, but still remains in the same column as the text. I can explain in more detail what the issues are when you start actively working on this.
